Metalama.Framework.Engine/CodeModel/Builders/NamedTypeBuilder.cs

```diff
@@ -246,7 +251,14 @@ public override SyntaxTree PrimarySyntaxTree
                         List<UsingDirectiveSyntax>(),
                         List<AttributeListSyntax>(),
                         List<MemberDeclarationSyntax>() ),
-                    path: this.FullName + ".cs",
+                    path: this.TypeParameters.Count > 0
+                        ? $"{this.FullName}`{this.TypeParameters.Count}.cs"
+                        : $"{this.FullName}.cs",
```

Comment:
Edge case: what happens on Windows if you have both `foo` and `Foo` types? Since C# is case sensitive, but the Windows filesystem isn't.

Though maybe that's not worth solving.

Reply:
There is deduplication of source files later in the pipeline, which results in `_2` suffix, etc.